src/engine/Operation.cpp

```diff
@@ -134,47 +138,112 @@ shared_ptr<const ResultTable> Operation::getResult(bool isRoot,
       // individual results, but that requires changes in each individual
       // operation, therefore we currently only perform this expensive
       // change in the DEBUG builds.
+      // This check doesn't make sense when the result has not been evaluated
+      // yet, so it should be moved into the operations eventually.
       AD_EXPENSIVE_CHECK(
+          result.isDataEvaluated() ||
           result.checkDefinedness(getExternallyVisibleVariableColumns()));
```

Comment:
We need to talk about this TODO of yours @joka921 
I don't quite get what exactly the comment is supposed to mean, and would like to discuss how it should be wrapped in your opinion